shared/utils/checkStatus.ts

```diff
@@ -0,0 +1,37 @@
+import { capitalize } from "./capitalize";
+import { Types } from "@requestnetwork/request-client.js";
+
+export const checkStatus = (request: Types.IRequestDataWithEvents | null) => {
+  const balance = BigInt(request?.balance?.balance ?? 0);
+  const expectedAmount = BigInt(request?.expectedAmount ?? 0);
+  const today = new Date();
+  const dueDate = new Date(request?.contentData?.paymentTerms?.dueDate);
```

Comment:
Agreed. The `dueDate` handling logic must be defensive because the `contentData` is arbitrary. Request Invoicing can't assume that the request was created with a well-formatted `contentData`

Reply:
@sstefdev also this one?

Reply:
This is added